modules/cabal-project.nix

```diff
@@ -136,5 +137,12 @@ in {
       type = nullOr (listOf unspecified);
       default = [];
     };
+
+    # Default shell arguments
+    shell = mkOption {
+      # TODO make this a submodule
+      type = unspecified;
```

Comment:
It would be nice to have module option docs here, to help users know how to use it.